src/status_im2/contexts/chat/messages/drawers/view.cljs

```diff
@@ -195,3 +198,72 @@
               :on-press            (fn []
                                      (rf/dispatch [:hide-bottom-sheet])
                                      (when on-press (on-press)))}]))]])))
+
+(defn get-reaction-authors-accessibility-id
+  [reaction-type]
+  (-> reaction-type
+      constants/reactions
+      name
+      (str "-reaction-count")))
+
+(defn contact-list-item-fn
+  [{:keys [from compressed-key]}]
+  (let [[primary-name secondary-name] (rf/sub [:contacts/contact-two-names-by-identity
+                                               from])
+        {:keys [ens-verified added?]} (rf/sub [:contacts/contact-by-address from])]
+    ^{:key (str compressed-key)}
+    [contact-list-item/contact-list-item
+     {:on-press #(rf/dispatch [:chat.ui/show-profile from])}
+     {:primary-name   primary-name
+      :secondary-name secondary-name
+      :public-key     from
+      :compressed-key compressed-key
+      :ens-verified   ens-verified
+      :added?         added?}]))
+
+(defn get-tabs-data
+  [reaction-authors selected-tab]
+  (mapv (fn [[reaction-type author-details]]
+          ^{:key (str reaction-type)}
+          {:id                  reaction-type
+           :style               style/tabs
+           :label               [rn/view {:style style/tab}
+                                 [quo/icon
+                                  (get constants/reactions reaction-type)
+                                  {:no-color        true
+                                   :container-style style/tab-icon}]
+                                 [quo/text
+                                  {:weight :medium
+                                   :size   :paragraph-1
+                                   :style  (style/tab-count (= @selected-tab
+                                                               reaction-type))}
+                                  (count author-details)]]
+           :accessibility-label (get-reaction-authors-accessibility-id reaction-type)})
+        reaction-authors))
+
+(defn reaction-authors-comp
+  [selected-tab reaction-authors show-reaction-author-list?]
+  (rn/use-effect (fn []
+                   (fn []
+                     (reset! show-reaction-author-list? false)))
+                 [reaction-authors])
+  [:<>
+   [quo/tabs
+    {:style          style/tabs
+     :size           28
+     :on-change      #(reset! selected-tab %)
+     :default-active @selected-tab
+     :data           (get-tabs-data reaction-authors selected-tab)}]
+   [rn/scroll-view
+    {:style {:height 320
+             :flex   1}}
+    (doall
```

Comment:
Isn't the `doall` unnecessary?

Reply:
I got a warning about not exhausting the collection so I used it.

Reply:
I just tried your branch and I don't get warnings. I actually don't understand why `doall` is needed here. Which tool is reporting this?

Reply:
Metro bundler's logs

Reply:
i see more and more usage of map in hiccup, but its better to use flat-list with render fn , or at least `for` it seems more natural for hiccup

Reply:
could you use flat-list pls